ext/socket/raddrinfo.c

```diff
@@ -318,17 +320,42 @@ rb_getaddrinfo(const char *node, const char *service,
     if (ret == 0)
         allocated_by_malloc = 1;
     else {
+        VALUE scheduler = rb_scheduler_current();
+
+        if (scheduler != Qnil && rb_scheduler_supports_address_resolve(scheduler)) {
+            VALUE host = rb_str_new_cstr(node);
+            VALUE ip_addresses_array = rb_scheduler_address_resolve(scheduler, host, Qnil);
+
+            long i, len = RARRAY_LEN(ip_addresses_array);
+            const VALUE *array_ptr = RARRAY_CONST_PTR(ip_addresses_array);
+            VALUE ip_address_str;
+            int ip_address_ret;
+            char *hostp;
+            char hbuf[NI_MAXHOST]; // ignored
+            int additional_flags = 0; // ignored
+
+            for(i=0; i<len; i++) {
+                ip_address_str = array_ptr[i];
+                hostp = host_str(ip_address_str, hbuf, sizeof(hbuf), &additional_flags);
+                ip_address_ret = numeric_getaddrinfo(hostp, service, hints, &ai);
```

Comment:
We are leveraging an existing function that converts numeric host (ipv4 or ipv6 address) with `service`(port) and `hints` (flags) into an addrinfo struct.